api/valor_api/backend/metrics/segmentation.py

```diff
@@ -202,27 +202,29 @@ def _compute_segmentation_metrics(
             "grouper_id_to_grouper_label_mapping"
         ][grouper_id]
 
-        ret.append(
-            IOUMetric(
-                label=grouper_label,
-                value=computed_iou_score,
+        if "IOU" in parameters.metrics:
+            ret.append(
+                IOUMetric(
+                    label=grouper_label,
+                    value=computed_iou_score,
+                )
             )
-        )
 
         ious_per_grouper_key[grouper_label.key].append(computed_iou_score)
 
     # aggregate IOUs by key
-    ret += [
-        mIOUMetric(
-            value=(
-                sum(iou_values) / len(iou_values)
-                if len(iou_values) != 0
-                else -1
-            ),
-            label_key=grouper_key,
-        )
-        for grouper_key, iou_values in ious_per_grouper_key.items()
-    ]
+    if "mIOU" in parameters.metrics:
```

Comment:
Are we not going to store the metrics that we computed? Is this desirable given the user could request these in the future? I feel that the computational cost of rerunning the evaluation to get a singular metric outweighs any memory cost of storing them even if they weren't requested in the moment.

I also feel that this stands for the detailed metrics. Why not add a `details` attribute to `schemas.Metric` and only populate the response with it if a flag is set?

Reply:
responded via slack. storing intermediary metrics isn't advantageous since a second `evaluate_` request with new parameters would trigger a second evaluation.

Reply:
changed this PR to store all metrics per our discussion yesterday